### PR TITLE
Add categories cards in Panel and enabled filter with categories

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

Add categories card in details panel add posibility of filter on them by selected Criterias in the Filter

https://www.loom.com/share/8e2bccb9c26946d6ae460e40af569f9b

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Add service and host category in configuration , then set a host or service with one of them.
Export and reload configuration
Go to the resources status menu and start to editing filter, to choose which category you want to display.

In details panel you can see a new card with the category set on the resources who looks like the groups's card.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by Jordan91 — APPROVED** [review]

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/DetailsChips.tsx [hunk 1]
```diff
@@ -22,27 +21,23 @@ const useStyles = makeStyles((theme) => ({
...
       <Grid item xs={12}>
         <Typography color="textSecondary" variant="body1">
-          {t(labelGroups)}
+          {t(title)}
         </Typography>
       </Grid>
-      {details?.groups?.map((group) => {
-        return <GroupChip group={group} key={group.id} type={groupType} />;
+      {groups?.map((group) => {
+        return <DetailsChip group={group} key={group.id} type={getType()} />;
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/DetailsChips.tsx:37]: You can extract it to a variable

**Review by Thebarda — COMMENTED** [review]
Don't forget to add translations on new labels

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -16875,3 +16875,12 @@ msgstr "Une erreur s'est produite lors de la récupération des catégories d'h
 
 msgid "BBDO protocol version enabled not compatible with this feature. Version needed '%s' or higher"
 msgstr "La version du protocole BBDO utilisée n'est pas compatible avec cette fonctionnalité. Version requise '%s' ou supérieure"
+
+msgid "Categories"
+msgstr "Catégories"
+
+msgid "Host category"
+msgstr "Catégories d'hôte"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16883]:
```suggestion
msgstr "Catégories d'hôtes"
```

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: www/front_src/src/Resources/Filter/Criterias/searchQueryLanguage/index.test.ts [hunk 3]
```diff
@@ -57,10 +57,28 @@ const parsedSearch = [
...
   },
-  { name: 'search', object_type: null, type: 'text', value: 'h.name:centreon' },
+  {
+    name: 'host_categories',
+    object_type: 'host_categories',
+    type: 'multi_select',
+    value: [{ id: 0, name: 'Linux' }],
+  },
+  {
+    isOptimizedModeEnabled: true,
```
Comment (bdauria) [inline www/front_src/src/Resources/Filter/Criterias/searchQueryLanguage/index.test.ts:67]: This property shouldn't be here in my opinion. Instead, we should have a mapping somewhere indicating which criteria should be displayed in optimized mode and be queried whenever needed before the actual display

**Review by bdauria — CHANGES_REQUESTED** [review]

**Comment by adr-mo** [comment]
Closing this PR, final review will be done on the final PR

**Pull request closed without merging**